plugins/coverimage.koplugin/main.lua

```diff
@@ -158,270 +190,526 @@ function CoverImage:onReaderReady(doc_settings)
+                        number = _("off")
+                    end
+                    return T(_("Maximum size of cached covers (%1MiB)"), number)
+                end,
+                help_text = _("If set to zero the cache size is unlimited.\nIf set to -1 the cache is disabled."),
+                checked_func = function()
+                    return self.cover_image_cache_maxsize >= 0
+                end,
+                callback = function(touchmenu_instance)
+                    self:sizeSpinner(touchmenu_instance, "cover_image_cache_maxsize", _("Cache size"), -1, 100, 5, self.cleanCache)
+                end,
+            },
+            self:menu_entry_set_path("cover_image_cache_path", _("Cover cache folder"), _("Current cache path:\n%1"),
+                ("Select a cache folder. The contents of the old folder will be migrated."), default_cache_path, true, false, self.migrateCache),
+            {
+                text = _("Clear cached covers"),
+                help_text_func = function()
+                    local cache_count, cache_size_KiB
+                        = self:getCacheFiles(self.cover_image_cache_path, self.cover_image_cache_prefix)
+                    return T(_("The cache contains %1 files and uses %2 MiB."), cache_count, math.floor((cache_size_KiB + 1023) / 1024))
+                end,
+                callback = function()
+                    UIManager:show(ConfirmBox:new{
+                        text =  _("Do you really want to clear the cover image cache?"),
+                        ok_text = _("Clear"),
+                        ok_callback = function()
+                            self:emptyCache()
+                        end,
+                    })
+                end,
+                keep_menu_open = true,
+            },
+        },
+    }
+end
+
+--[[--
+Menu entry for setting an specific G_reader_setting key for a path/file
+
+@string key is the G_reader_setting key which is used and changed
+@string title shown in the menu
+@string help shown in the menu
+@string info shown in the menu (if containing %1, the value of the key is shown)
+@string the default value
+@bool folder_only sets if only folders can be selected
+@bool new_file sets if a new filename can be entered
+@function migrate a callback for example moving the folder contents
+]]
+function CoverImage:menu_entry_set_path(key, title, help, info, default, folder_only, new_file, migrate)
+    return {
+        text = title,
+        help_text_func = function()
+            local text = self[key]
+            text = text ~= "" and text or _("not set")
+            return T(help, text)
+        end,
+        checked_func = function()
+            return isFileOk(self[key]) or (isPathAllowed(self[key]) and folder_only)
+        end,
+        callback = function(touchmenu_instance)
+            UIManager:show(ConfirmBox:new{
+                text = info,
+                ok_text = _("Yes"),
+                cancel_text = _("No"),
```

Comment:
This sounds like a rephrasing of the default Cancel/OK that doesn't add anything?

Reply:
You are right -> I have removed the texts.